shell/platform/android/io/flutter/plugin/editing/TextInputPlugin.java

```diff
@@ -71,13 +83,40 @@ public void clearClient() {
                 clearTextInputClient();
             }
         });
+        this.platformViewsController = platformViewsController;
+        platformViewsController.attachTextInputPlugin(this);
```

Comment:
As you mentioned, I think we need symmetrical behavior with `attachTextInputPlugin()`. It's fine to wait for me to invoke that behavior in teh new embedding, but for historical reference purposes I would recommend adding that capacity here. At the moment we're introducing an impossibility to clean up that `attach` relationship.

I'm not sure what the naming convention is elsewhere in the embedding, it might be `destroy`, but we should probably go ahead and introduce such a method here for symmetry.

Reply:
In the current embedding the TextInputPlugin is created with a FlutterView and is never explicitly destroyed so I'm not sure where I'd add that. In the new embedding I believe there's a natural place to do that.

Reply:
I think what we'd want is a destroy() method beneath the constructor in `TextInputPlugin` and that method invokes `platformViewsController.detachTextInputPlugin()`. That works, right?

Reply:
Yes, the only caveat is that right now it seems like we're "destroying" the TextInputPlugin by restarting the input connection here:
https://github.com/flutter/engine/blob/2bc2cee67b8001913e4fe34f7c7e8e7776c2fd7c/shell/platform/android/io/flutter/embedding/android/FlutterView.java#L535

I guess we'd want to move the equivalent logic into this new `destroy` method, probably not as-is(seems like the mechanism to kill the input connection is baked into FlutterView's `onCreateInputConnection`, wouldn't we want the TextInputPlugin to contain all the logic for managing an input connection?)